src/App.js

```diff
@@ -3,18 +3,84 @@ import Card from './Card';
 import { useState } from 'react';
 
 function App() {
-  const [checked, setChecked] = useState(false);
+  const [cardState, setCardState] = useState({
+    checked: false,
+    isEditable: false
+  });
+
+  const [cardText, setCardText] = useState({
+    headerText: 'Caption',
+    tempHeader: 'Caption',
+    bodyText: 'text...',
+    tempBody: 'text...'
+  });
+
   const changeStyleHandler = () => {
-    setChecked(!checked);
-  }
+    setCardState({
+      ...cardState,
+      checked: !cardState.checked,
+    });
+  };
+
+  const openEditModeHandler = () => {
+    setCardState({
+      checked: false,
+      isEditable: !cardState.isEditable
+    });
+  };
+
+  const headerTextChangeHandler = e => {
+    setCardText({
+      ...cardText,
+      headerText: e.target.value
+    });
+  };
+
+  const bodyTextChangeHandler = e => {
+    setCardText({
+      ...cardText,
+      bodyText: e.target.value
+    });
+  };
+
+  const saveChangesHandler = () => {
+    setCardState({
+      ...cardState,
+      isEditable: !cardState.isEditable
+    });
+    setCardText({
+      ...cardText,
+      tempHeader: cardText.headerText,
+      tempBody: cardText.bodyText
+    })
+  };
+
+  const cancelChangesHandler = () => {
+    setCardState({
+      ...cardState,
+      isEditable: !cardState.isEditable
+    });
+    setCardText({
+      ...cardText,
+      headerText: cardText.tempHeader,
+      bodyText: cardText.tempBody
+    });
+  };
+
   return (
     <div className="App">
       <Header />
-      <Card 
-        caption="Caption" 
-        text="Text..." 
+      <Card
+        bodyText={cardText.bodyText}
         onChange={changeStyleHandler}
-        checked={checked}
+        checked={cardState.checked}
+        openEditMode={openEditModeHandler}
+        isEditable={cardState.isEditable}
+        onSaveChanges={saveChangesHandler}
+        onCancelChanges={cancelChangesHandler}
+        headerTextChangeHandler={headerTextChangeHandler}
+        headerText={cardText.headerText}
```

Comment:
You can actually pass `cardData `object and destruct it in Card component if needed.